### PR TITLE
feat: use new gpt-3.5-turbo model

### PR DESCRIPTION
**Pull request by scottarver**
the tokenizer that was just added doesnt have support for the new model yet, so still using the told model and 🤞 they are very similar.

I used it to generate its own commit message.

### CODE REVIEW

**Review by Helpmylyfe82 — APPROVED** [review]

**Comment by salomartin** [comment]
This works well for me and seems to write better commit messages :-)

**Comment by juliavdkris** [comment]
Working pretty well for me!

I also made a slight change to the prompt: "'Write a high-level, insightful and concise Git commit message in a complete sentence in present tense for the following diff without prefacing it with anything:'", which at least in my testing made it give slightly less detail-focused commit messages :)

File: src/utils/openai.ts [hunk 1]
```diff
@@ -94,7 +98,7 @@ export const generateCommitMessage = async (
 
 		return deduplicateMessages(
 			completion.choices
-				.map(choice => sanitizeMessage(choice.text!)),
+				.map(choice => sanitizeMessage(choice.message?.content ?? 'No Commit Message')),
```
Comment (privatenumber) [inline src/utils/openai.ts:101]: We should throw an error instead of falling back to a made up commit message.

File: src/utils/openai.ts [hunk 2]
```diff
@@ -60,8 +60,9 @@ const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 
 const getPrompt = (locale: string, diff: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the following diff without prefacing it with anything, the response must be in the language ${locale}:\n${diff}`;
 
-const model = 'text-davinci-003';
-const encoder = encodingForModel(model);
+const model = 'gpt-3.5-turbo';
+// TODO: update for the new gpt-3.5 model
+const encoder = encodingForModel('text-davinci-003');
```
Comment (privatenumber) [inline src/utils/openai.ts:65]: Was waiting for the new one to be released but I guess there's no point in blocking the feature.

**Comment by privatenumber** [comment]
Thanks!

**Pull request merged**